labeling/keyword_labeler.go

```diff
@@ -1,42 +1,60 @@
 package labeling
 
 import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
 	"strings"
 
-	bsky "github.com/bluesky-social/indigo/api/bsky"
+	appbsky "github.com/bluesky-social/indigo/api/bsky"
 )
 
-// simple record labeling (without pre-fetched blobs)
-type SimplePostLabeler interface {
-	LabelPost(p bsky.FeedPost) []string
-}
-type SimpleActorProfileLabeler interface {
-	LabelActorProfile(ap bsky.ActorProfile) []string
-}
-
 type KeywordLabeler struct {
-	keywords []string
-	value    string
+	Keywords []string `json:"keywords"`
+	Value    string   `json:"value"`
 }
 
 func (kl KeywordLabeler) LabelText(txt string) []string {
 	txt = strings.ToLower(txt)
-	for _, word := range kl.keywords {
+	for _, word := range kl.Keywords {
 		if strings.Contains(txt, word) {
-			return []string{kl.value}
+			return []string{kl.Value}
 		}
 	}
 	return []string{}
 }
 
-func (kl KeywordLabeler) LabelPost(p bsky.FeedPost) []string {
+func (kl KeywordLabeler) LabelPost(p appbsky.FeedPost) []string {
 	return kl.LabelText(p.Text)
 }
 
-func (kl KeywordLabeler) LabelActorProfile(ap bsky.ActorProfile) []string {
+func (kl KeywordLabeler) LabelProfile(ap appbsky.ActorProfile) []string {
 	txt := ap.DisplayName
 	if ap.Description != nil {
 		txt += *ap.Description
 	}
 	return kl.LabelText(txt)
 }
+
+func LoadKeywordFile(fpath string) ([]KeywordLabeler, error) {
+
+	var kwl []KeywordLabeler
+
+	jsonFile, err := os.Open(fpath)
+	if err != nil {
+		return kwl, fmt.Errorf("failed to load JSON file: %v", err)
+	}
+	defer jsonFile.Close()
+
+	raw, err := io.ReadAll(jsonFile)
+	if err != nil {
+		return kwl, fmt.Errorf("failed to load JSON file: %v", err)
```

Comment:
definitely not blocking, but its always a bit awkward when i see a non-nil value along with an error return. It signals to me that you intend to use the error *and* the value, when the intention here is to ignore the value in the error case. It draws the eye unnecessarily